[BUG] 1.4.4: compile time warnings

**Describe the bug**
On compile time are emitted several warnings.

**To Reproduce**
Just compile latest srt using latest gcc using LTO.
Part of the warnings are related to use openssl 3.0.0 (those for now are less important)
Here is example script which should produce all warnings:
```shell
CFLAGS='-O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none'
+ LDFLAGS='-Wl,-z,relro -Wl,--as-needed -Wl,--gc-sections -Wl,-z,now -flto=auto -flto-partition=none -fuse-linker-plugin'
CC=/usr/bin/gcc
AR=/usr/bin/gcc-ar
NM=/usr/bin/gcc-nm
RANLIB=/usr/bin/gcc-ranlib
export CFLAGS CXXFLAGS FFLAGS FCFLAGS LDFLAGS CC CXX FC AR NM RANLIB
/usr/bin/cmake -D ENABLE_GETNAMEINFO=ON -D ENABLE_STATIC=OFF -D ENABLE_UNITTESTS=ON -D USE_GNUSTL=OFF -D USE_OLD_ADD_METHOD=OFF
```

**Expected behaviour**
No compile time warnings especially those related to use LTO.

**Screenshots**
Summary stats:
```console
[tkloczko@barrel SPECS]$ rpmbuild -ba --with check --with failing_tests srt.spec --quiet 2>&1 | grep -- -W | sed 's/.*\[//; s/\]//' | sort | uniq -c | sort -nr
      5 -Wdeprecated-declarations
      1 -Wodr
      1 -Wlto-type-mismatch
```
Warnings extracted from stderr:
```console
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.c: In function 'crysprOpenSSL_AES_SetKey':
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.c:43:9: warning: 'AES_set_encrypt_key' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
   43 |         if (AES_set_encrypt_key(kstr, kstr_len * 8, aes_key)) {
      |         ^~
In file included from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.h:24,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-config.h:9,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr.h:39,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/hcrypt.h:52,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.c:20:
/usr/include/openssl/aes.h:51:5: note: declared here
   51 | int AES_set_encrypt_key(const unsigned char *userKey, const int bits,
      |     ^~~~~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.c:48:9: warning: 'AES_set_decrypt_key' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
   48 |         if (AES_set_decrypt_key(kstr, kstr_len * 8, aes_key)) {
      |         ^~
In file included from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.h:24,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-config.h:9,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr.h:39,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/hcrypt.h:52,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.c:20:
/usr/include/openssl/aes.h:54:5: note: declared here
   54 | int AES_set_decrypt_key(const unsigned char *userKey, const int bits,
      |     ^~~~~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.c: In function 'crysprOpenSSL_AES_CtrCipher':
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.c:119:27: warning: 'AES_encrypt' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  119 |                           inlen, aes_key, iv, ctr, &blk_ofs, (block128_f) AES_encrypt);
      |                           ^~~~~
In file included from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.h:24,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-config.h:9,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr.h:39,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/hcrypt.h:52,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.c:20:
/usr/include/openssl/aes.h:57:6: note: declared here
   57 | void AES_encrypt(const unsigned char *in, unsigned char *out,
      |      ^~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.c: In function 'crysprOpenSSL_KmWrap':
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.c:154:5: warning: 'AES_wrap_key' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  154 |     return(((seklen + HAICRYPT_WRAPKEY_SIGN_SZ) == (unsigned int)AES_wrap_key(kek, NULL, wrap, sek, seklen)) ? 0 : -1);
      |     ^~~~~~
In file included from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.h:24,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-config.h:9,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr.h:39,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/hcrypt.h:52,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.c:20:
/usr/include/openssl/aes.h:97:5: note: declared here
   97 | int AES_wrap_key(AES_KEY *key, const unsigned char *iv,
      |     ^~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.c: In function 'crysprOpenSSL_KmUnwrap':
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.c:166:5: warning: 'AES_unwrap_key' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  166 |     return(((wraplen - HAICRYPT_WRAPKEY_SIGN_SZ) == (unsigned int)AES_unwrap_key(kek, NULL, sek, wrap, wraplen)) ? 0 : -1);
      |     ^~~~~~
In file included from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.h:24,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-config.h:9,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr.h:39,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/hcrypt.h:52,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.4.4/haicrypt/cryspr-openssl.c:20:
/usr/include/openssl/aes.h:101:5: note: declared here
  101 | int AES_unwrap_key(AES_KEY *key, const unsigned char *iv,
      |     ^~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/apps/verbose.hpp:30:7: warning: type 'struct Log' violates the C++ One Definition Rule [-Wodr]
   30 | class Log
      |       ^
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/apps/verbose.hpp:30:7: note: a different type is defined in another translation unit
   30 | class Log
      |       ^
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/apps/verbose.hpp:34:10: note: the first difference of corresponding definitions is field 'lockline'
   34 |     bool lockline = false;
      |          ^
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/apps/verbose.hpp:30:7: note: a type with different number of fields is defined in another translation unit
   30 | class Log
      |       ^
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/apps/verbose.hpp:56:5: warning: type of '__dt_comp ' does not match original declaration [-Wlto-type-mismatch]
   56 |     ~Log();
      |     ^
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/apps/verbose.cpp:39:5: note: '__dt_comp ' was previously declared here
   39 |     Log::~Log()
      |     ^
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/apps/verbose.cpp:39:5: note: code may be misoptimized unless '-fno-strict-aliasing' is used
```
Openssl 3.0.0 deprecation warnings can be for now skipped.
**Desktop (please provide the following information):**
 - OS: Linux/x86_64
 - SRT Version / commit ID: 1.4.4

**Additional context**
N/A

Hi @kloczek 
Thanks for reporting!
Warnings related to OpenSSL 3.0 are described in #2141.

The remaining to be considered/fixed within this issue are:

```shell
verbose.hpp:30:7: warning: type 'struct Log' violates the C++ One Definition Rule [-Wodr]
   30 | class Log
      |       ^
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/apps/verbose.hpp:30:7: note: a different type is defined in another translation unit
   30 | class Log
      |       ^
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/apps/verbose.hpp:34:10: note: the first difference of corresponding definitions is field 'lockline'
   34 |     bool lockline = false;
      |          ^
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/apps/verbose.hpp:30:7: note: a type with different number of fields is defined in another translation unit
   30 | class Log
      |       ^
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/apps/verbose.hpp:56:5: warning: type of '__dt_comp ' does not match original declaration [-Wlto-type-mismatch]
   56 |     ~Log();
      |     ^
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/apps/verbose.cpp:39:5: note: '__dt_comp ' was previously declared here
   39 |     Log::~Log()
      |     ^
/home/tkloczko/rpmbuild/BUILD/srt-1.4.4/apps/verbose.cpp:39:5: note: code may be misoptimized unless '-fno-strict-aliasing' is used
```

Just update for 1.5.0 and stil lI see LTO related warnings
```console
[tkloczko@devel-g2v SPECS]$ rpmbuild -ba --with check --with failing_tests srt.spec --quiet 2>&1 | grep -- \\[-W | sed 's/.*\[//; s/\]//' | sort | uniq -c | sort -nr
      5 -Wdeprecated-declarations
      2 -Walloc-size-larger-than=
      1 -Wuninitialized
      1 -Wodr
      1 -Wlto-type-mismatch
```
In case of `-Wlto-type-mismatch` all what needs to be done is sync declaration of what is in .hpp header file and in .cpp.
Curent list of warnings
<details>

```console
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/haicrypt/cryspr-openssl.c: In function 'crysprOpenSSL_AES_SetKey':
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/haicrypt/cryspr-openssl.c:43:9: warning: 'AES_set_encrypt_key' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
   43 |         if (AES_set_encrypt_key(kstr, kstr_len * 8, aes_key)) {
      |         ^~
In file included from /home/tkloczko/rpmbuild/BUILD/srt-1.5.0/haicrypt/cryspr-openssl.h:24,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.5.0/haicrypt/cryspr-config.h:9,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.5.0/haicrypt/cryspr.h:39,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.5.0/haicrypt/hcrypt.h:52,
                 from /home/tkloczko/rpmbuild/BUILD/srt-1.5.0/haicrypt/cryspr-openssl.c:20:
/usr/include/openssl/aes.h:51:5: note: declared here
   51 | int AES_set_encrypt_key(const unsigned char *userKey, const int bits,
      |     ^~~~~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/haicrypt/cryspr-openssl.c:48:9: warning: 'AES_set_decrypt_key' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
   48 |         if (AES_set_decrypt_key(kstr, kstr_len * 8, aes_key)) {
      |         ^~
/usr/include/openssl/aes.h:54:5: note: declared here
   54 | int AES_set_decrypt_key(const unsigned char *userKey, const int bits,
      |     ^~~~~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/haicrypt/cryspr-openssl.c: In function 'crysprOpenSSL_AES_CtrCipher':
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/haicrypt/cryspr-openssl.c:119:27: warning: 'AES_encrypt' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  119 |                           inlen, aes_key, iv, ctr, &blk_ofs, (block128_f) AES_encrypt);
      |                           ^~~~~
/usr/include/openssl/aes.h:57:6: note: declared here
   57 | void AES_encrypt(const unsigned char *in, unsigned char *out,
      |      ^~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/haicrypt/cryspr-openssl.c: In function 'crysprOpenSSL_KmWrap':
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/haicrypt/cryspr-openssl.c:154:5: warning: 'AES_wrap_key' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  154 |     return(((seklen + HAICRYPT_WRAPKEY_SIGN_SZ) == (unsigned int)AES_wrap_key(kek, NULL, wrap, sek, seklen)) ? 0 : -1);
      |     ^~~~~~
/usr/include/openssl/aes.h:97:5: note: declared here
   97 | int AES_wrap_key(AES_KEY *key, const unsigned char *iv,
      |     ^~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/haicrypt/cryspr-openssl.c: In function 'crysprOpenSSL_KmUnwrap':
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/haicrypt/cryspr-openssl.c:166:5: warning: 'AES_unwrap_key' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  166 |     return(((wraplen - HAICRYPT_WRAPKEY_SIGN_SZ) == (unsigned int)AES_unwrap_key(kek, NULL, sek, wrap, wraplen)) ? 0 : -1);
      |     ^~~~~~
/usr/include/openssl/aes.h:101:5: note: declared here
  101 | int AES_unwrap_key(AES_KEY *key, const unsigned char *iv,
      |     ^~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/srtcore/packet.cpp: In constructor 'srt::CPacket::CPacket()':
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/srtcore/packet.cpp:180:25: warning: member 'srt::CPacket::m_nHeader' is used uninitialized [-Wuninitialized]
  180 |     m_iSeqNo((int32_t&)(m_nHeader[SRT_PH_SEQNO])),
      |                         ^~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/srtcore/queue.cpp: In member function 'init':
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/srtcore/queue.cpp:106:31: warning: argument 1 value '18446744073709551615' exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]
  106 |         tempu = new CUnit[size];
      |                               ^
/usr/include/c++/12/new:128:26: note: in a call to allocation function 'operator new []' declared here
  128 | _GLIBCXX_NODISCARD void* operator new[](std::size_t) _GLIBCXX_THROW (std::bad_alloc)
      |                          ^
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/srtcore/queue.cpp: In member function 'increase':
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/srtcore/queue.cpp:172:31: warning: argument 1 value '18446744073709551615' exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]
  172 |         tempu = new CUnit[size];
      |                               ^
/usr/include/c++/12/new:128:26: note: in a call to allocation function 'operator new []' declared here
  128 | _GLIBCXX_NODISCARD void* operator new[](std::size_t) _GLIBCXX_THROW (std::bad_alloc)
      |                          ^
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/apps/verbose.hpp:30:7: warning: type 'struct Log' violates the C++ One Definition Rule [-Wodr]
   30 | class Log
      |       ^
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/apps/verbose.hpp:30:7: note: a different type is defined in another translation unit
   30 | class Log
      |       ^
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/apps/verbose.hpp:34:10: note: the first difference of corresponding definitions is field 'lockline'
   34 |     bool lockline = false;
      |          ^
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/apps/verbose.hpp:30:7: note: a type with different number of fields is defined in another translation unit
   30 | class Log
      |       ^
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/apps/verbose.hpp:56:5: warning: type of '__dt_comp ' does not match original declaration [-Wlto-type-mismatch]
   56 |     ~Log();
      |     ^
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/apps/verbose.cpp:39:5: note: '__dt_comp ' was previously declared here
   39 |     Log::~Log()
      |     ^
/home/tkloczko/rpmbuild/BUILD/srt-1.5.0/apps/verbose.cpp:39:5: note: code may be misoptimized unless '-fno-strict-aliasing' is used
```
</details>

Is it possible to reopen this thicket?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1383,7 +1383,6 @@ if (ENABLE_APPS)
 		# srt-multiplex temporarily blocked
 		#srt_add_application(srt-multiplex ${VIRTUAL_srtsupport})
 		srt_add_application(srt-tunnel ${VIRTUAL_srtsupport})
-		target_compile_definitions(srt-tunnel PUBLIC -DSRT_ENABLE_VERBOSE_LOCK)
 	endif()
 
 	if (ENABLE_TESTING)
@@ -1421,7 +1420,6 @@ if (ENABLE_APPS)
 
 		srt_add_testprogram(srt-test-relay)
 		srt_make_application(srt-test-relay)
-		target_compile_definitions(srt-test-relay PUBLIC -DSRT_ENABLE_VERBOSE_LOCK)
 
 		srt_add_testprogram(srt-test-multiplex)
 		srt_make_application(srt-test-multiplex)

--- a/apps/verbose.cpp
+++ b/apps/verbose.cpp
@@ -9,14 +9,13 @@
  */
 
 #include "verbose.hpp"
+#include <mutex>
 
 namespace Verbose
 {
     bool on = false;
     std::ostream* cverb = &std::cerr;
-#if SRT_ENABLE_VERBOSE_LOCK
     std::mutex vlock;
-#endif
 
     Log& Log::operator<<(LogNoEol)
     {
@@ -28,19 +27,16 @@ namespace Verbose
         return *this;
     }
 
-#if SRT_ENABLE_VERBOSE_LOCK
     Log& Log::operator<<(LogLock)
     {
         lockline = true;
         return *this;
     }
-#endif
 
     Log::~Log()
     {
         if (on && !noeol)
         {
-#if SRT_ENABLE_VERBOSE_LOCK
             if (lockline)
             {
                 // Lock explicitly, as requested, and wait for the opportunity.
@@ -48,7 +44,7 @@ namespace Verbose
             }
             else if (vlock.try_lock())
             {
-                // Successfully locked, so unlock immediately, locking wasn't requeted.
+                // Successfully locked, so unlock immediately, locking wasn't required.
                 vlock.unlock();
             }
             else
@@ -62,15 +58,12 @@ namespace Verbose
                 vlock.lock();
                 vlock.unlock();
             }
-#endif
             (*cverb) << std::endl;
-#if SRT_ENABLE_VERBOSE_LOCK
 
             // If lockline is set, the lock was requested and WAS DONE, so unlock.
             // Otherwise locking WAS NOT DONE.
             if (lockline)
                 vlock.unlock();
-#endif
         }
     }
 }

--- a/apps/verbose.cpp
+++ b/apps/verbose.cpp
@@ -9,13 +9,13 @@
  */
 
 #include "verbose.hpp"
-#include <mutex>
+#include "sync.h" // srt::sync
 
 namespace Verbose
 {
     bool on = false;
     std::ostream* cverb = &std::cerr;
-    std::mutex vlock;
+    srt::sync::Mutex vlock;
 
     Log& Log::operator<<(LogNoEol)
     {
@@ -44,7 +44,7 @@ namespace Verbose
             }
             else if (vlock.try_lock())
             {
-                // Successfully locked, so unlock immediately, locking wasn't required.
+                // Successfully locked, so unlock immediately, locking wasn't requested.
                 vlock.unlock();
             }
             else

--- a/apps/verbose.hpp
+++ b/apps/verbose.hpp
@@ -12,9 +12,6 @@
 #define INC_SRT_VERBOSE_HPP
 
 #include <iostream>
-#if SRT_ENABLE_VERBOSE_LOCK
-#include <mutex>
-#endif
 
 namespace Verbose
 {
@@ -23,19 +20,15 @@ extern bool on;
 extern std::ostream* cverb;
 
 struct LogNoEol { LogNoEol() {} };
-#if SRT_ENABLE_VERBOSE_LOCK
 struct LogLock { LogLock() {} };
-#endif
 
 class Log
 {
     bool noeol = false;
-#if SRT_ENABLE_VERBOSE_LOCK
     bool lockline = false;
-#endif
 
     // Disallow creating dynamic objects
-    void* operator new(size_t);
+    void* operator new(size_t) = delete;
 
 public:
 
@@ -50,9 +43,7 @@ public:
     }
 
     Log& operator<<(LogNoEol);
-#if SRT_ENABLE_VERBOSE_LOCK
     Log& operator<<(LogLock);
-#endif
     ~Log();
 };
 
@@ -99,8 +90,6 @@ inline void Verb(Args&&... args)
 
 // Manipulator tags
 static const Verbose::LogNoEol VerbNoEOL;
-#if SRT_ENABLE_VERBOSE_LOCK
 static const Verbose::LogLock VerbLock;
-#endif
 
 #endif

